src/porepy/numerics/ad/operators.py

```diff
@@ -1861,99 +1936,35 @@ def copy_common_sub_tags(self) -> None:
-        domains = [var.domain for var in self.sub_vars]
-        # Verify that all domains of of the same type
-        assert all(isinstance(d, pp.Grid) for d in domains) or all(
-            isinstance(d, pp.MortarGrid) for d in domains
-        )
-        return domains
+        return [var.domain for var in self.sub_vars]
 
     @property
     def size(self) -> int:
         """Returns the total size of the mixed-dimensional variable
         by summing the sizes of sub-variables."""
         return sum([v.size for v in self.sub_vars])
 
-    def previous_timestep(self) -> MixedDimensionalVariable:
-        """Return a representation of this mixed-dimensional variable on the previous
-        time step.
-
-        Raises:
-            ValueError:
-                If the variable is a representation of the previous iteration,
-                previously set by :meth:`~previous_iteration`.
-
-            NotImplementedError:
-                If the variable is already a representation of the previous time step.
-                Currently, we support creating only one previous time step.
-
-        Returns:
-            A representation of this merged variable on the previous time
-            iteration, with its ``prev_iter`` attribute set to ``True``.
-
-        """
-
-        new_subs = [var.previous_timestep() for var in self.sub_vars]
-        new_var = MixedDimensionalVariable(new_subs)
-        new_var.prev_time = True
-        # Assign self as the original variable.
-        new_var.original_variable = self
-        return new_var
-
-    def previous_iteration(self) -> MixedDimensionalVariable:
-        """Return a representation of this mixed-dimensional variable on the previous
-        iteration.
-
-        Raises:
-            ValueError:
-                If the variable is a representation of the previous time step,
-                previously set by :meth:`~previous_timestep`.
-
-            NotImplementedError:
-                If the variable is already a representation of the previous time
-                iteration. Currently, we support creating only one previous iteration.
-
-        Returns:
-            A representation of this merged variable on the previous
-            iteration, with its ``prev_iter`` attribute set to ``True``
-
-        """
-        new_subs = [var.previous_iteration() for var in self.sub_vars]
-        new_var = MixedDimensionalVariable(new_subs)
-        new_var.prev_iter = True
-        # Assign self as the original variable.
-        new_var.original_variable = self
-        return new_var
-
-    def copy(self) -> "MixedDimensionalVariable":
-        """Copy the mixed-dimensional variable.
-
-        Returns:
-            A shallow copy should be sufficient here; the attributes are not expected to
-            change.
-
-        """
-        return copy.deepcopy(self)
+    def parse(self, mdg: pp.MixedDimensionalGrid) -> Any:
```

Comment:
MD variables are not parsed, but their sub vars

Reply:
I should comment at this point, that the `TimeDependentArray` parses itself with a grid-wise order.
So we have the order of equations/dofs implemented in multiple places in porepy. @keileg @IvarStefansson @OmarDuran 
One solution would be, to give the `EquationSystem` as argument to parse, and implement a method in `EquationSystem` returning the row indices per grid.
Then an MD variable could also parse itself, and the row indices would be very useful for e.g. optimizing the sparsity pattern